src/main/java/uk/gov/hmcts/reform/pip/account/management/service/AuthorisationService.java

```diff
@@ -0,0 +1,75 @@
+package uk.gov.hmcts.reform.pip.account.management.service;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.pip.account.management.database.UserRepository;
+import uk.gov.hmcts.reform.pip.account.management.errorhandling.exceptions.NotFoundException;
+import uk.gov.hmcts.reform.pip.account.management.model.PiUser;
+import uk.gov.hmcts.reform.pip.model.account.Roles;
+
+import java.util.UUID;
+
+import static uk.gov.hmcts.reform.pip.model.LogBuilder.writeLog;
+import static uk.gov.hmcts.reform.pip.model.account.Roles.ALL_NON_RESTRICTED_ADMIN_ROLES;
+import static uk.gov.hmcts.reform.pip.model.account.Roles.ALL_NON_THIRD_PARTY_ROLES;
+
+@Service("authorisationService")
+@Slf4j
+public class AuthorisationService {
+    @Autowired
+    UserRepository userRepository;
+
+    public boolean userCanDeleteAccount(UUID userId, UUID adminUserId) {
+        boolean isAuthorised = isAuthorisedRole(userId, adminUserId);
+
+        if (!isAuthorised) {
+            log.error(writeLog(
+                String.format("User with ID %s is forbidden to remove user with ID %s", adminUserId, userId)
+            ));
+        }
+        return isAuthorised;
+    }
+
+    public boolean userCanUpdateAccount(UUID userId, UUID adminUserId) {
+        if (adminUserId != null && adminUserId.equals(userId)) {
+            log.error(writeLog(
+                String.format("User with ID %s is forbidden to update their owned account", userId)
```

Comment:
I think this should be 'own' rather than 'owned'

Reply:
Corrected typo